src/runtime/storageNG/storage-proxy.ts

```diff
@@ -191,7 +208,7 @@ export class StorageProxy<T extends CRDTTypeRecord> {
           if (!this.synchronized) {
             // If we didn't think we were synchronized but the operation applied cleanly,
             // then actually we were synchronized after all. Tell the handle that.
-            this.setSynchronized();
+            this.setSynchronized(initialModel);
```

Comment:
why the initialModel instead of the model after operations are applied?

Reply:
The sync gets delivered first so it needs to have the state of the model before the updates.